### PR TITLE
feat(users): add approve and reject methods to User

### PR DESCRIPTION
**Pull request by bgamari**
Closes https://github.com/python-gitlab/python-gitlab/issues/1604.

Follow-up to https://github.com/python-gitlab/python-gitlab/pull/2064.

### CODE REVIEW

**Comment by bgamari** [comment]
I have amended the commit message as suggested.

Unfortunately, it's not clear how to test this robustly given that the GitLab instance the tests are being run against must be configured to have new account approvals enabled. However, I am using this patch locally in managing spam account creation on https://gitlab.haskell.org/

**Comment by nejch** [comment]
> I have amended the commit message as suggested.
> 
> Unfortunately, it's not clear how to test this robustly given that the GitLab instance the tests are being run against must be configured to have new account approvals enabled. However, I am using this patch locally in managing spam account creation on https://gitlab.haskell.org/

Thanks for the quick response @bgamari! 

We run integration tests against a gitlab container, but for unit tests, we just mock the responses (see file linked above). We usually just use the example responses from the API docs for the mocked response. I guess we should document this in our contributing guide a bit more, thanks for the feedback! :grin: 

In this case, this would just be "Examples responses" in https://docs.gitlab.com/ee/api/users.html#approve-user - we only really need to test the success case.

It's not really the best example, but you could check https://github.com/python-gitlab/python-gitlab/blob/main/tests/unit/objects/test_users.py#L63-L72 to see how the fixtures are used because it's so similar (though ideally we'd have an individual response as a fixture). Let me know if you're happy to work on that, otherwise we can merge and do this after. Thanks again :)

**Review by nejch — CHANGES_REQUESTED** [review]

**Comment by bgamari** [comment]
Thanks for the feedback, @nejch. I'll add a few unit tests.

File: docs/gl_objects/users.rst
Comment (JohnVillalovos) [inline docs/gl_objects/users.rst:113]: How do we "reject" someone?

Comment (bgamari) [reply docs/gl_objects/users.rst:113]: I've added a mention of `reject()` as well.

File: gitlab/v4/objects/users.py
Comment (JohnVillalovos) [inline gitlab/v4/objects/users.py:314]: Minor fix:

```suggestion
            GitlabUserRejectError: If the user could not be rejected
```

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

**Review by nejch — APPROVED** [review]

**Comment by nejch** [comment]
@bgamari thanks again for this, I rebased your branch and squashed the commit co-authored by John to make pre-commit happy. @JohnVillalovos LGTM any other concerns on your side?

**Review by JohnVillalovos — APPROVED** [review]

**Comment by JohnVillalovos** [comment]
> @bgamari thanks again for this, I rebased your branch and squashed the commit co-authored by John to make pre-commit happy. @JohnVillalovos LGTM any other concerns on your side?

@nejch LGTM. I'm good to merge if you are.

**Pull request merged**
